stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+import aspose.threed as a3d
+import argparse
+import os
+import bpy
+import json
+import shutil
+import textures as Textures # custom module created to store texture info
+
+# example run
+# python3.10 stl_to_tscn.py input.json
+
+# uv map functions from bpy
+UV_MAPS = {
+    "smart": bpy.ops.uv.smart_project,
+    "cube": bpy.ops.uv.cube_project,
+    "cylinder": bpy.ops.uv.cylinder_project,
+    "sphere": bpy.ops.uv.sphere_project,
+    "unwrap": bpy.ops.uv.unwrap
+}
+
+class TscnGen:    
+    def __init__(self, input_folder, output_folder, scale):
+        self.input_folder = input_folder
+        self.output_folder = output_folder
+
+        # all the .obj, .glb, .jpg (textures)
+        self.ext_resource = ""
+        self.ext_resource_id = 1
+
+        # SpatialMaterials (construction of textures)
+        self.sub_resource = ""
+        self.sub_resource_id = 1
+
+        # MeshInstances and collision scenes (.glb)
+        self.nodes = f'[node name="{output_folder}" type="Spatial"]\n'
+        if scale != 1:
+            self.nodes += f'transform = Transform( {scale}, 0, 0, 0, {scale}, 0, 0, 0, {scale}, 0, 0, 0 )\n'
+        self.nodes += "\n" # add extra new line to prepare for next node
+
+        # keep track of textures so we don't generate them more than once
+        self.texture_index = dict()
+
+    def add_texture(self, texture):
+        # don't generate textures more than once
+        if texture not in self.texture_index:
```

Comment:
```suggestion
        # don't generate textures more than once
        if texture in self.texture_index:
            w = Warning(f'{texture} has already been generated.')
            raise w
        else:
```
You can consider raising some warning here instead to notify users that they are attempting to add the texture more than once.

Reply:
I don't think this will be an issue because it is possible that users will want to use the same texture on different `.stl` files. Not regenerating textures is primarily to avoid doing more work than we have to (if the texture has already been imported, we can just reuse it instead of generating it again).